modules/application/src/main/java/edu/northeastern/truthtree/adapter/basicInfo/BasicInfoDBAdapter.java

```diff
@@ -148,22 +156,81 @@ public JSONArray getBasicCitiesInfo(int startValue, int endValue) {
    * @return JSONArray that contains states that are within the provided range.
    */
   @Override
-  public JSONArray getBasicCountiesInfo(int startValue, int endValue) {
-    JSONArray response = getPopulationData(startValue, endValue, 1);
-    return addPopulationToBasicInfo(countiesData, response);
+  public PageDTO<LocationDTO> getBasicCountiesInfo(int page, int startValue, int endValue)
+          throws IOException {
+    return getPopulationData(page, startValue, endValue, countiesData, 1);
+  }
+
+  private PageDTO<LocationDTO> getPopulationData(int currentPage, int startValue, int endValue,
+                                 Map<Long, Object> basicInfoMap, int typeCode)
+          throws IOException {
+    String url = UriComponentsBuilder
+            .fromHttpUrl(POPULATION_RANGE_URL)
+            .queryParam("attributeId", POPULATION_ID)
+            .queryParam("year", 2016)
+            .queryParam("from", startValue)
+            .queryParam("to", endValue)
+            .queryParam("currentPage", currentPage)
+            .queryParam("typeCode", typeCode)
+            .queryParam("pageSize", 50)
+            .toUriString();
+
+    ObjectMapper mapper = new ObjectMapper();
+    ResponseEntity<String> response = restTemplate.getForEntity(url, String.class);
```

Comment:
Can you add RESTTemplate call in URLUtils class? I understand for this GET call, your response is not a DTO, but you can make it generic so that this type of method definition is reusable everywhere.

Reply:
I would rather not, the `RestTemplate` call is 1-2 line at most, create a wrapper method for it is just not worth it